### ISSUE TITLE
Clear an 'og-group-content' cache tag whenever group content changes

### ISSUE THREAD

**Issue by pfrenssen** [Gizra/og#216]
In my project I am displaying a listing of group content belonging to a specific group. I would like to invalidate the cache of this listing whenever new content is added to the group, or the existing content changes.

For the moment I have implemented this with two hook implementations in my project. I'm wondering if this would perhaps be useful to add to OG itself? I'm not sure how we can make this easily discoverable though, we currently don't offer any views or so that would benefit from having this cache tag.

```
/**
 * Implements hook_entity_update().
 */
function collection_entity_update(Drupal\Core\Entity\EntityInterface $entity) {
  // If group content is created or updated, invalidate the group content cache
  // tags for each of the groups this group content belongs to. This allows
  // group listings to be cached effectively. The cache tag format is
  // 'og-group-content:{group entity type}:{group entity id}'.
  if (Og::isGroupContent($entity->getEntityTypeId(), $entity->bundle())) {
    $tags = [];
    foreach (\Drupal::service('og.membership_manager')->getGroupIds($entity) as $entity_type => $ids) {
      foreach ($ids as $id) {
        $tags[] = "$entity_type:$id";
      }
    }
    Cache::invalidateTags(Cache::buildTags('og-group-content', $tags));
  }
}

/**
 * Implements hook_entity_delete().
 */
function collection_entity_delete(Drupal\Core\Entity\EntityInterface $entity) {
  collection_entity_update($entity);
}
```

**Comment by amitaibu** [comment]
This is pretty sweet. We can add a `Recent group content` view to demo it (+ it's going to be a handy view)

**Comment by pfrenssen** [comment]
OK thanks! Will make a PR for it.

**Comment by pfrenssen** [comment]
> We can add a Recent group content view to demo it (+ it's going to be a handy view)

I just started on this but this is actually impossible, a View is bound to a single entity type, and group content can be of multiple entity types!

I can make a simple block though that presents the most recent group content in an unordered list.

**Comment by amitaibu** [comment]
We can do recent group content _nodes_ - this is going to be the majority of cases.

**Comment by pfrenssen** [comment]
OK!

**Comment by pfrenssen** [comment]
Have to abandon the Views idea again, I created a default argument that provides the group from OgContext but I'm stuck now on the contextual filter. The relationship between a group and its group content is done using the group audience fields, but these can be given arbitrary names and there can be more than one group audience field. I lack the knowledge and time to figure out if this is even possible and how to do it. I'm going to provide a simple block instead.

I will make a PR to contribute the default argument plugin I wrote, that might come in handy for people that are building their own views.
